lms/views/decorators/h_api.py

```diff
@@ -121,73 +121,54 @@ def wrapper(request, jwt, context=None):
             return wrapped(request, jwt)
 
         context = context or request.context
-        get_param = functools.partial(_get_param, request)
-
-        tool_consumer_instance_guid = get_param("tool_consumer_instance_guid")
-        context_id = get_param("context_id")
-
-        group = models.CourseGroup.get(
-            request.db, tool_consumer_instance_guid, context_id
-        )
-
-        assert group is not None, (
-            "create_course_group() should always have "
-            "been run successfully before this "
-            "function gets called, so group should "
-            "never be None."
-        )
 
         request.find_service(name="hapi").post(
-            f"groups/{group.pubid}/members/{context.h_userid}"
+            f"groups/{context.h_pubid}/members/{context.h_userid}"
         )
 
         return wrapped(request, jwt)
 
     return wrapper
 
 
-def _maybe_create_group(context, request):
-    """Create a Hypothesis group for the LTI course, if one doesn't exist."""
+def _upsert_group(context, request):
```

Comment:
This function is named `_upsert_group` but nothing in it makes use of the `PUT` endpoint (`PATCH` and `POST` only). Are you intending to extend it so it uses `PUT` in some cases? Or should it be renamed? The logic here seems to differ from your pseudo-code in the PR description in that it's using `POST` instead of `PUT` if the group doesn't exist and the request's user is a teacher.

I mean, this is fine, but I want to understand where the app is going to make use of the `PUT`.

Reply:
Corrected this to use `PUT`